core/src/test/java/com/linecorp/armeria/server/logging/AccessLogFormatsTest.java

```diff
@@ -255,6 +276,115 @@ public void requestLogAvailabilityException() {
+        final Duration duration = Duration.ofMillis(1000000000L);
+
+        log.endRequest();
+        when(log.requestStartTimeMillis()).thenReturn(requestStartTime.toEpochMilli());
+        when(log.requestDurationNanos()).thenReturn(duration.toNanos());
+
+        format = AccessLogFormats.parseCustom("%{requestStartTimeMillis}L " +
+                                              "%{requestEndTimeMillis}L " +
+                                              "%{requestDurationMillis}L");
+        assertThat(AccessLogger.format(format, log)).isEqualTo(
+                String.join(" ",
+                            String.valueOf(requestStartTime.toEpochMilli()),
+                            String.valueOf(requestStartTime.plus(duration).toEpochMilli()),
+                            String.valueOf(duration.toMillis())));
+
+        format = AccessLogFormats.parseCustom("\"%{requestCause}L\"");
+        assertThat(AccessLogger.format(format, log)).isEqualTo("\"-\"");
+        when(log.requestCause()).thenReturn(new IllegalArgumentException("detail_message"));
+        assertThat(AccessLogger.format(format, log))
+                .isEqualTo('"' + IllegalArgumentException.class.getSimpleName() + ": detail_message\"");
+
+        format = AccessLogFormats.parseCustom("%{responseStartTimeMillis}L " +
+                                              "%{responseEndTimeMillis}L " +
+                                              "%{responseDurationMillis}L");
+        // No values.
+        assertThat(AccessLogger.format(format, log)).isEqualTo("- - -");
+
+        log.startResponse();
+        log.endResponse();
+
+        final Instant responseStartTime = Instant.ofEpochMilli(2000000000L);
+
+        when(log.responseStartTimeMillis()).thenReturn(responseStartTime.toEpochMilli());
+        when(log.responseDurationNanos()).thenReturn(duration.toNanos());
+
+        assertThat(AccessLogger.format(format, log)).isEqualTo(
+                String.join(" ",
+                            String.valueOf(responseStartTime.toEpochMilli()),
+                            String.valueOf(responseStartTime.plus(duration).toEpochMilli()),
+                            String.valueOf(duration.toMillis())));
+
+        format = AccessLogFormats.parseCustom("\"%{responseCause}L\"");
+        assertThat(AccessLogger.format(format, log)).isEqualTo("\"-\"");
+        when(log.responseCause()).thenReturn(new IllegalArgumentException());
+        assertThat(AccessLogger.format(format, log))
+                .isEqualTo('"' + IllegalArgumentException.class.getSimpleName() + '"');
+    }
+
+    @Test
+    public void timestamp() {
+        final DummyRequestContext ctx = new DummyRequestContext();
+        final DefaultRequestLog log = spy(new DefaultRequestLog(ctx));
+
+        log.startRequest(channel, SessionProtocol.H2, "www.example.com");
+        // To generate the same datetime string always.
+        final long requestStartTimeMillis = 0L;
```

Comment:
How about changing to some value which is not `0`? something like `19830725`?

Reply:
Ok. But I don't like `19830725`. :stuck_out_tongue_winking_eye: